### ISSUE TITLE
Excluding LF in 'allowed-comment-char' is confusing

### ISSUE THREAD

**Issue by arp242** [toml-lang/toml#995]
I'm adding a test to toml-test for "Relax comment parsing; most control characters are again permitted.", and implementing it in my parser, and I'm rather confused what 0x0a (LF) is excluded. You need to support this if you want CRLF, no?

There's a long thread in #924 about this, but to be honest I stopped reading after a few comments – people shouldn't need to read entire threads to understand the specification.

I think the intent is that "CR" and "CRLF" are allowed, but "LF" isn't? That is what it does, I think, but it's really not very obvious IMHO.

At the very least there should be a comment in the ABNF, and the phrasing in toml.md ("that could cause problems during editing or processing: U+0000, and U+000A to U+000D") could also be improved. Now it's too easy to be misinterpreted as "only CR newlines are allowed and LF is forbidden, full stop", which isn't what was intended.

**Comment by arp242** [comment]
Actually 0x0d (CR) is also included ... this is even more confusing. What's the point of including that?

**Comment by arp242** [comment]
And a whole bunch of tests fail because now the allowed character set for strings and comments differ, and there was always the assumption those were the same.

I really wish we'd have "fix toml-test" as a requirement for *any* change.

[None of the hypotheticals](https://github.com/toml-lang/toml/issues/567#issuecomment-568269120) are even solved, because you still need to parse-by-character and exclude 0x00, 0x0b, 0x0c, you can still invalidate documents with a control character.

All for a problem no one actually reported. IMHO it's a completely pointless change that doesn't change anything for anyone, doesn't solve any practical problem, and is just ivory tower wankery and argueing for the point of argueing.

But I'm the one mucking about with toml-test and dealing with the churn (16 failing tests...)

So great, thanks.

**Comment by ChristianSi** [comment]
> I think the intent is that "CR" and "CRLF" are allowed, but "LF" isn't?

No, CR and CRLF are not allowed *in* a comment, they *end* the line and hence the comment.

As for why LF, though not a valid line terminator, is not allowed in comments, i think this commented explains it best: https://github.com/toml-lang/toml/pull/924#issuecomment-1278831310.

**Comment by ChristianSi** [comment]
As for being as permissive as possible in comments and not erroring out on users because of harmless control characters, I think that's a good thing and certainly not something we should apologize for.

**Comment by arp242** [comment]
Yeah, but the way it's phrased now is confusing, IMHO. Or at least: I was confused by it, but maybe that's just on my 🙃

Anyway, I just so happened to submit a PR to revert this; while this is entirely a fixable issue as such, I don't see the point in spending any time on this as I don't see how it fixes anything.

**Comment by ChristianSi** [comment]
Hmm, if the language is confusing, suggestions to improve it are certainly welcome!

**Comment by ChristianSi** [comment]
As for the "doesn't fix anything", I don't think that's true? Consider the case where somebody includes a harmless control char in a comment, and the TOML parser balk out on them with "invalid document, can't process!" That's unexpected and not particularly user-friendly, I'd say, considering that comments are supposed to be ignored anyway?

So I'd doing away with such error message is a net win, isn't it?

CF is probably the only control char that's not "harmless", since it could possibly lead to the user seeing a different document than the parser, which is why the special treatment for it makes sense (and was adapted).

**Comment by arp242** [comment]
LF was already forbidden, so this PR didn't change anything about that:

> Control characters other than tab (U+0000 to U+0008, U+000A to U+001F, U+007F) are not permitted in comments.

**Comment by ChristianSi** [comment]
It didn't change that, and quite wisely so, I'd say, but it did allow the other control characters in.

**Comment by arp242** [comment]
> thus making TOML more robust and user-friendly.

Who is running in to problems with this? Have people reported problems?
